multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -276,4 +281,13 @@ private boolean shouldUpdateCredentials(CloudServiceInstanceExtended service, Cl
             throw e;
         }
     }
+
+    private boolean shouldUpdateSyslogUrl(CloudServiceInstance service, String existingSyslogUrl) {
+        if (!service.isUserProvided()) {
+            return false;
+        }
+        String syslogDrainUrl = ObjectUtils.defaultIfNull(service.getSyslogDrainUrl(), "");
```

Comment:
If `existingSyslogUrl` is not null, this line is redundant.

Reply:
This line makes sure that, in the case of  `existingSyslogUrl`  being an empty string, it wouldn't be compared to a null  `syslogDrainUrl`.

Reply:
In this context, is there a difference between an empty-string URL and a null one?

Reply:
No there isn't ,  but since we're comparing with equals, null and an empty string are considered different.  
From testing when updating a service with no defined `syslog-drain-url` parameter, it would seem that  `existingService.getSyslogDrainUrl()` returns an empty string when it isn't set, while `service.getSyslogDrainUrl()` returns null.